lib/amo-client.js

```diff
@@ -110,7 +110,12 @@ Client.prototype.waitForSignedAddon = function(statusUrl, opt) {
     function checkSignedStatus() {
       self.get({url: statusUrl}).then(function(result) {
         var data = result[1];
-        if (data.processed) {
+
+        var failedValidation = !data.valid;
+        // The add-on passes validation and all files have been created.
+        var passedValidation = data.valid && data.reviewed;
+
+        if (data.processed && (failedValidation || passedValidation)) {
```

Comment:
I find this a bit weird. Expanding `failedValidation` and `passedValidation` would simplify to `data.processed && (!data.valid || data.reviewed)` Which I think is just as clear. Maybe with a comment.